src/dotnet/APIView/APIViewWeb/Repositories/NotificationManager.cs

```diff
@@ -0,0 +1,103 @@
+        }
+
+        private string GetHtmlContent(CommentModel comment, ReviewModel review)
+        {
+            var uri = new Uri($"{_endpoint}/Assemblies/Review/{review.ReviewId}#{comment.ElementId}");
+            var sb = new StringBuilder();
+            sb.Append(GetContentHeading(comment, true));
+            sb.Append("<br><br>");
+            sb.Append($"In <a href='{uri.ToString()}'>{comment.ElementId}</a>:");
+            sb.Append("<br><br>");
+            sb.Append(comment.Comment);
+            return sb.ToString();
+        }
+
+        private string GetPlainTextContent(CommentModel comment)
+        {
+            var sb = new StringBuilder();
+            sb.Append(GetContentHeading(comment, false));
+            sb.Append("\r\n");
+            sb.Append(comment.Comment);
+            return sb.ToString();
+        }
+
+        private static string GetContentHeading(CommentModel comment, bool includeHtml) =>
+            $"{(includeHtml ? $"<b>{comment.Username}</b>" : $"{comment.Username}")} commented on this review at {comment.TimeStamp}";
+
+        public async Task NotifySubscribersOnNewRevisionAsync(ReviewRevisionModel revision, ClaimsPrincipal user)
+        {
+            var review = revision.Review;
+            var uri = new Uri($"{_endpoint}/Assemblies/Review/{review.ReviewId}");
+            var plainTextContent = $"A new revision, {revision.Name}," +
+                $" was uploaded by {revision.Author} at {revision.CreationDate}";
+            var htmlContent = $"A new revision, <a href='{uri.ToString()}'>{revision.Name}</a>," +
+                $" was uploaded by <b>{revision.Author}</b> at {revision.CreationDate}";
+            await SendEmailsAsync(review, user, plainTextContent, htmlContent);
+        }
+        private async Task SendEmailsAsync(ReviewModel review, ClaimsPrincipal user, string plainTextContent, string htmlContent)
+        {
+            if (review.Subscribers.Count == 0)
+            {
+                return;
+            }
+            var client = new SendGridClient(_sendGridKey);
+            var from = new EmailAddress(FROM_ADDRESS, FROM_NAME);
+            var initiatingUserEmail = GetUserEmail(user);
+            SendGridMessage msg = MailHelper.CreateMultipleEmailsToMultipleRecipients(
+                from,
+                review.Subscribers.ToList()
+                    .Where(e => e != initiatingUserEmail) // don't include the initiating user in the email
+                    .Select(e => new EmailAddress(e))
+                    .ToList(),
+                Enumerable.Repeat(review.Name, review.Subscribers.Count).ToList(),
+                plainTextContent,
+                htmlContent,
+                Enumerable.Repeat(new Dictionary<string, string>(), review.Subscribers.Count).ToList());
+            await client.SendEmailAsync(msg);
+        }
+
+        public async Task UpsertReviewAsync(ReviewModel review) =>
```

Comment:
Turn it into `SubscribeAsyn(User user, string reviewId)` and `UnsubscribeAsync(User user, string reviewId)` so you don't have to go through 2 managers to get subscribed or unsubscribed.